cvat/apps/engine/frame_provider.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (C) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import math
+from io import BytesIO
+
+from cvat.apps.engine.media_extractors import VideoReader, ZipReader
+from cvat.apps.engine.models import DataChoice
+from cvat.apps.engine.mime_types import mimetypes
+
+
+class FrameProvider():
```

Comment:
Just a comment for future discussion. I generally like the class but prefer to change its interface:

- QUALITY_ORIGINAL=100, QUALITY_COMPRESSED=0
- get_preview(self)
- get_chunk(self, n, quality=QUALITY_ORIGINAL)
- get_frame(self, n, quality=QUALITY_ORIGINAL)
- get_frames(self, quality=QUALITY_ORIGINAL)


The main idea behind the change is to keep the API stable if we generate compressed frames on the fly with a quality. Thus we will be able to call get_frame(self, n, 75).